core/src/storage/device.py

```diff
@@ -114,10 +125,25 @@ def is_passphrase_enabled() -> bool:
     return common.get_bool(_NAMESPACE, _USE_PASSPHRASE)
 
 
+def set_passphrase_enabled(enable: bool) -> None:
+    common.set_bool(_NAMESPACE, _USE_PASSPHRASE, enable)
+    if not enable:
+        set_passphrase_always_on_device(False)
+
+
 def get_homescreen() -> Optional[bytes]:
     return common.get(_NAMESPACE, _HOMESCREEN, True)  # public
 
 
+def set_homescreen(homescreen: bytes) -> None:
+    if len(homescreen) > HOMESCREEN_MAXSIZE:
+        raise ValueError  # homescreen too large
+    if homescreen[:8] == b"TOIf\x90\x00\x90\x00" or homescreen == b"":
+        common.set(_NAMESPACE, _HOMESCREEN, homescreen, True)  # public
```

Comment:
So this PR solves https://github.com/trezor/trezor-firmware/issues/1118 correct?

Reply:
No, this checks the TOIF header but not the body. But this is the place to do it.